ext/sockets/sockets.c

```diff
@@ -2825,8 +2826,8 @@ PHP_FUNCTION(socket_addrinfo_lookup)
 		} ZEND_HASH_FOREACH_END();
 	}
 
-	if (getaddrinfo(ZSTR_VAL(hostname), service, &hints, &result) != 0) {
-		RETURN_FALSE;
+	if ((ret = getaddrinfo(ZSTR_VAL(hostname), service, &hints, &result)) != 0) {
+		RETURN_LONG(ret);
```

Comment:
This is a BC break that might be quite problematic because users often do `=== false` check so I don't think this acceptable in minor. You will need RFC for this.

Reply:
minor version :)